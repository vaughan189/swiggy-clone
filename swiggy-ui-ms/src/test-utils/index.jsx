import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { theme } from '../theme';

const AllTheProviders = ({ children, mocks = [] }) => {
  return (
    <MockedProvider mocks={mocks} addTypename={false}>
        {children}
    </MockedProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
  mocks: PropTypes.arrayOf(PropTypes.shape({})),
};

AllTheProviders.defaultProps = {
  mocks: [],
};

const customRender = (ui, options) => {
  const { wrapperProps = {}, ...customOptions } = options || {};
  return render(ui, {
    wrapper: (props) => {
      return <AllTheProviders {...props} {...wrapperProps} />;
    },
    ...customOptions,
  });
};

// re-export everything
export * from '@testing-library/react';
export * from './withTestRouter';

// override render method
export { customRender as render };
