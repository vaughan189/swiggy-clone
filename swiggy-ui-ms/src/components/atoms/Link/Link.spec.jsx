import React from 'react';
import { render, withTestRouter } from '../../../test-utils';

import Link from '.';

describe('Link', () => {
  it('should render correctly', () => {
    const { asFragment } = render(withTestRouter(<Link href="/">This is dummy link</Link>));
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with naked true', () => {
    const { asFragment } = render(
      withTestRouter(
        <Link href="/" naked>
          This is dummy link
        </Link>
      )
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
