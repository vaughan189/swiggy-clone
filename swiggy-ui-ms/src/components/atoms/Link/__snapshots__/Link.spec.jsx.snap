// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="/"
  >
    This is dummy link
  </a>
</DocumentFragment>
`;

exports[`Link should render correctly with naked true 1`] = `
<DocumentFragment>
  <a
    class=""
    href="/"
  >
    This is dummy link
  </a>
</DocumentFragment>
`;
